File: Exercism/c/pangram/src/pangram.c
#include <string.h>
#include <ctype.h>
#include "pangram.h"

bool is_pangram(const char* sentence) {
    if (!sentence)
        return false;

    bool marker[26];
    for (int i = 0 ; i < 26 ; ++i)
        marker[i] = false;

    for (size_t i = 0 ; i < strlen(sentence) ; ++i) {
        if (isalpha(sentence[i]))
            marker[tolower(sentence[i]) - 97] = true;
    }

    for (int i = 0 ; i < 26 ; ++i) {
        if (!marker[i])
            return false;
    }

    return true;
}

File: Exercism/ruby/pangram/pangram.rb
# frozen_string_literal: true

require 'set'

# Pangram ...
class Pangram
  def self.pangram?(sentence)
    letters = Set[]

    sentence.downcase.chars do |x|
      letters.add(x) if x =~ /[a-z]/
    end

    letters.size == 26
  end
end
